### PR TITLE
additional xhr type fixes

### PR DESCRIPTION
**Pull request by erwinmombay**
base diff is against https://github.com/ampproject/amphtml/pull/4239

### CODE REVIEW

File: src/service/xhr-impl.js [hunk 1]
```diff
@@ -121,6 +124,7 @@ export class Xhr {
     return this.fetch_(input, opt_init).catch(reason => {
       user.assert(false, 'Fetch failed %s: %s', input,
           reason && reason.message);
+      return response;
```
Comment (erwinmombay) [inline src/service/xhr-impl.js]: @dvoytenko please review this line specifically if this is the intent or not

Comment (jridgewell) [inline src/service/xhr-impl.js]: `response` isn't defined.

Comment (erwinmombay) [inline src/service/xhr-impl.js]: whoops, fixing

File: src/service/xhr-impl.js [hunk 2]
```diff
@@ -139,6 +139,9 @@ export class Xhr {
             ` ${ALLOW_SOURCE_ORIGIN_HEADER} header`);
       }
       return response;
+    }).catch(reason => {
+      user.assert(false, 'Fetch failed %s: %s', input,
```
Comment (erwinmombay) [inline src/service/xhr-impl.js]: @dvoytenko @jridgewell if i'm understanding correctly the original intent of the catch was just to throw additional info on the error message so it should be fine to move this here at the bottom. (fixes the type error of compiler thinking the catch should return something which the next `then` handler should receive a its paramter)

Comment (jridgewell) [inline src/service/xhr-impl.js]: Yup, these are equivalent.

Comment (erwinmombay) [inline src/service/xhr-impl.js]: sorry switched this out with `.then(succHandler, errHandler)` which i believe should be more similar to the original behavior. (just making sure you saw the latest change)

Comment (jridgewell) [inline src/service/xhr-impl.js]: Whichever is currently written is the one I reviewed.

File: src/service/xhr-impl.js [hunk 3]
```diff
@@ -485,7 +489,8 @@ export class FetchResponse {
                'responseType was not "arraybuffer"');
     dev.assert(!this.bodyUsed, 'Body already used');
     this.bodyUsed = true;
-    return Promise.resolve(this.xhr_.response);
+    return /** @type {!Promise<ArrayBuffer>} */ (
```
Comment (dvoytenko) [inline src/service/xhr-impl.js]: `ArrayBuffer` is nullable?

File: src/styles.js [hunk 4]
```diff
@@ -189,7 +189,7 @@ function styleLoaded(doc, style) {
 /**
  * Insert the element in the root after the element named after or
  * if that is null at the beginning.
- * @param {!Element} root
+ * @param {!Element|ShadowRoot} root
```
Comment (dvoytenko) [inline src/styles.js]: `!ShadowRoot`?

**Comment by dvoytenko** [comment]
Couple of questions, but otherwise LGTM

File: src/styles.js (hunk 4 above)
Comment (erwinmombay) [inline src/styles.js]: done.

File: src/service/xhr-impl.js (hunk 3 above)
Comment (erwinmombay) [inline src/service/xhr-impl.js]: added !

Comment (erwinmombay) [inline src/service/xhr-impl.js]: although it currently is nullable https://github.com/ampproject/amphtml/blob/master/src/service/xhr-impl.js#L481 and we don't assert anywhere that it isn't. i believe xhr.response is nullable by default right? (on failures?)

Comment (dvoytenko) [inline src/service/xhr-impl.js]: Typically we do errors. But it's fine if it's nullable if it's a valid case. In that case - let's document this well.

Comment (erwinmombay) [inline src/service/xhr-impl.js]: ive went ahead and added an assert for existence

Comment (dvoytenko) [inline src/service/xhr-impl.js]: Sg. LGTM

**Pull request merged**